### ISSUE TITLE
Server side websocket ReadAsync doesn't return when cancelled on Linux

### ISSUE THREAD

**Issue by dmpriso** [dotnet/corefx#30267]
When passing a ` ` `CancellationToken` ` ` to a ` ` ` System.Net.Websockets.WebSocket` ` `  which originated in a ` ` `HttpListenerContext.AcceptWebSocket` ` ` call, so a server-side websocket, and later cancelling that CancellationToken, the WebSocket's receive operation doesn't return until 30 seconds have passed.

I have created a small gist to repo that issue:
https://gist.github.com/dmpriso/df9b00b06f8dbb48f3748e42372212f6

When executing on Ubuntu 16.04, it fails as described, firing an ` ` ` OperationCancelledException` ` ` as expected but only after 30 seconds have passed.

On windows, the behaviour is as expected - after cancelling, the read operation returns immediately with an ` ` `OperationCancelledException` ` `

**Comment by wfurt** [comment]
I can run the repro and see the outcome. I think this essentially depends on #24430

**Comment by karelz** [comment]
Let's fold it into the master issue then:
Duplicate of #24430
